R/glimpse.R

```diff
@@ -98,3 +96,12 @@ format_v.list <- function(x) {
 
 #' @export
 format_v.character <- function(x) encodeString(x, quote = '"')
+
+#' @export
+format_v.factor <- function(x) {
+  if (any(grepl(",", x, fixed = TRUE))) {
+    encodeString(as.character(x), quote = '"')
```

Comment:
Does this mean that all factor levels will be quoted if at least one level has a comma?

Reply:
Yes, I think this makes sense because we want the quotation to be consistent across levels.

This way, if the user sees that one level is quoted and another is not, they know that the quoted level has a real quotation mark, i.e.

```
> cat(format_v(factor(c("foo", '"bar"'))))
foo "bar"
```